### PR TITLE
[Security] Fixes for PassportInterface deprecation and UserInterface token users

### PR DESCRIPTION
**Pull request by wouterj**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | sort of
| New feature?  | no
| Deprecations? | yes
| Tickets       | -
| License       | MIT
| Doc PR        | -

Some small fixes for changes done in recent PRs for security, caught while working on #41613

### CODE REVIEW

**Review by chalasr — APPROVED** [review]
Thanks, I missed those remaining PassportInterface occurrences. Just one minor comment

File: src/Symfony/Component/Security/Http/Authenticator/Passport/Passport.php [hunk 1]
```diff
@@ -59,6 +58,34 @@ public function getUser(): UserInterface
         return $this->user;
     }
 
+    /**
+     * @return $this
+     */
+    public function addBadge(BadgeInterface $badge): PassportInterface
```
Comment (chalasr) [inline src/Symfony/Component/Security/Http/Authenticator/Passport/Passport.php:64]:
```suggestion
    public function addBadge(BadgeInterface $badge): self
```

to be updated to `: static` on 6.0

Comment (wouterj) [reply src/Symfony/Component/Security/Http/Authenticator/Passport/Passport.php:64]: Unfortunately, this breaks PHP <7.4 https://3v4l.org/Nf1MJ

**Comment by wouterj** [comment]
> Thanks, I missed those remaining PassportInterface occurrences.

We all did while reviewing the PR :) Thanks for the quick reviews

File: src/Symfony/Component/Security/Http/Authenticator/Passport/Passport.php (hunk 1 above)
Comment (chalasr) [reply src/Symfony/Component/Security/Http/Authenticator/Passport/Passport.php:64]: Argh, of course

**Comment by chalasr** [comment]
Thank you @wouterj.

**Pull request merged**